pkg/version/version.go

```diff
@@ -75,14 +75,25 @@ func MakeInfoFromStrCheck(curVer string) (*Info, error) {
 // IsEqualOrNewer returns true if the version in the Vdb is is equal or newer
 // than the given version
 func (i *Info) IsEqualOrNewer(inVer string) bool {
-	inVerMajor, inVerMinor, inVerPatch, ok := parseVersion(inVer)
+	comp, ok := parseVersion(inVer)
 	if !ok {
 		panic(fmt.Sprintf("could not parse input version: %s", inVer))
 	}
-	res := i.compareVersion(inVerMajor, inVerMinor, inVerPatch)
+	res := i.compareVersion(comp)
 	return res != compareSmaller
 }
 
+// HasEqualOrNewerHotfix checks if both versions have the same major, minor, patch numbers, and
+// if hotfix number in source version is equal or newer than the one in target version
+func (i *Info) HasEqualOrNewerHotfix(inVer string) bool {
+	comp, ok := parseVersion(inVer)
+	if !ok {
+		panic(fmt.Sprintf("could not parse input version: %s", inVer))
+	}
+	res := i.compareVersion(comp)
+	return res == compareEqual && i.VdbHotfix >= comp.VdbHotfix
```

Comment:
why not allow a newer major/minor? looks like this is only used by isOnlineUpgradeSupported which should accept 24.3-2 and >= 24.4, right?